scripts/test.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+
+set -eu
+set -o pipefail
+
+readonly PROG_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+readonly STACK_DIR="$(cd "${PROG_DIR}/.." && pwd)"
+readonly OUTPUT_DIR="${STACK_DIR}/build"
+readonly OUTPUT_DIR_NODEJS16="${STACK_DIR}/build-nodejs-16"
+readonly OUTPUT_DIR_NODEJS18="${STACK_DIR}/build-nodejs-18"
+
+# shellcheck source=SCRIPTDIR/.util/tools.sh
+source "${PROG_DIR}/.util/tools.sh"
+
+# shellcheck source=SCRIPTDIR/.util/print.sh
+source "${PROG_DIR}/.util/print.sh"
+
+function main() {
+  local clean
+  clean="false"
+  while [[ "${#}" != 0 ]]; do
+    case "${1}" in
+      --help|-h)
+        shift 1
+        usage
+        exit 0
+        ;;
+
+      --clean|-c)
+        shift 1
+        clean="true"
+        ;;
+
+      "")
+        # skip if the argument is empty
+        shift 1
+        ;;
+
+      *)
+        util::print::error "unknown argument \"${1}\""
+    esac
+  done
+
+  tools::install
+
+  if [[ "${clean}" == "true" ]]; then
+    util::print::title "Cleaning up preexisting stack archives..."
+    rm -rf "${OUTPUT_DIR}"
+    rm -rf "${OUTPUT_DIR_NODEJS16}"
+    rm -rf "${OUTPUT_DIR_NODEJS16}"
+  fi
+
+  if ! [[ -f "${OUTPUT_DIR}/build.oci" ]] || ! [[ -f "${OUTPUT_DIR}/run.oci" ]] || ! [[ -f "${OUTPUT_DIR_NODEJS16}/run.oci" ]] || ! [[ -f "${OUTPUT_DIR_NODEJS18}/run.oci" ]]; then
+    util::print::title "Creating stack..."
+    "${STACK_DIR}/scripts/create.sh"
+  fi
+
+  tests::run
+}
+
+function usage() {
+  cat <<-USAGE
+test.sh [OPTIONS]
+
+Runs acceptance tests against the stack. Uses the OCI images
+${STACK_DIR}/build/build.oci
+and
+${STACK_DIR}/build/run.oci
+and
+${STACK_DIR}/build-nodejs-16/run.oci
+and
+${STACK_DIR}/build-nodejs-18/run.oci
+if they exist. Otherwise, first runs create.sh to create them.
```

Comment:
This is a nice-to-have, but it would be good if we could run the tests just for one of the stacks at a time as well. If I'm making changes to the Node.js 18 image for example, I would appreciate being able to isolate the tests to that one stack. Open to having this implemented down the line though if you want.

Reply:
@mhdawson can you file an issue for this if its not going to happen in this PR?

Reply:
Sure, issue created here - https://github.com/paketo-community/ubi-base-stack/issues/5